File: src/cli/dpkg.go
package cli

import (
	"strings"
)

// Dpkg wraps dpkg utility commands
type Dpkg struct {
	all            bool
	logFile        *string
	suppressErrors bool
}

// NewDpkg creates a new Dpkg command builder
func NewDpkg() *Dpkg {
	return &Dpkg{
		suppressErrors: true,
	}
}

// All configures all packages
func (d *Dpkg) All(value bool) *Dpkg {
	d.all = value
	return d
}

// LogFile sets log file path
func (d *Dpkg) LogFile(path string) *Dpkg {
	d.logFile = &path
	return d
}

// SuppressErrors suppresses errors
func (d *Dpkg) SuppressErrors(value bool) *Dpkg {
	d.suppressErrors = value
	return d
}

// Configure generates dpkg --configure command
func (d *Dpkg) Configure() string {
	parts := []string{"dpkg"}
	if d.all {
		parts = append(parts, "--configure", "-a")
	} else {
		parts = append(parts, "--configure")
	}
	if d.logFile != nil {
		parts = append(parts, ">"+quote(*d.logFile))
	}
	return strings.Join(parts, " ")
}

// Divert generates dpkg-divert command
func (d *Dpkg) Divert(path string, quiet, local, rename bool, action string) string {
	parts := []string{"dpkg-divert"}
	if quiet {
		parts = append(parts, "--quiet")
	}
	if local {
		parts = append(parts, "--local")
	}
	if rename {
		parts = append(parts, "--rename")
	}
	parts = append(parts, action, quote(path))
	return strings.Join(parts, " ") + ""
}

File: src/CLI/Dpkg.cs
using System;
using System.Linq;

namespace Enva.CLI;

public class Dpkg
{
    private bool all;
    private string? logFile;
    private bool suppressErrors = true;

    public static Dpkg NewDpkg()
    {
        return new Dpkg();
    }

    public Dpkg All(bool value)
    {
        all = value;
        return this;
    }

    public Dpkg LogFile(string path)
    {
        logFile = path;
        return this;
    }

    public Dpkg SuppressErrors(bool value)
    {
        suppressErrors = value;
        return this;
    }

    public string Configure()
    {
        var parts = new System.Collections.Generic.List<string> { "dpkg" };
        if (all)
        {
            parts.Add("--configure");
            parts.Add("-a");
        }
        else
        {
            parts.Add("--configure");
        }
        if (!string.IsNullOrEmpty(logFile))
        {
            parts.Add($">{Quote(logFile)}");
        }
        return string.Join(" ", parts);
    }

    public string Divert(string path, bool quiet, bool local, bool rename, string action)
    {
        var parts = new System.Collections.Generic.List<string> { "dpkg-divert" };
        if (quiet)
        {
            parts.Add("--quiet");
        }
        if (local)
        {
            parts.Add("--local");
        }
        if (rename)
        {
            parts.Add("--rename");
        }
        parts.Add(action);
        parts.Add(Quote(path));
        return string.Join(" ", parts);
    }

    private string Quote(string s)
    {
        if (s.Contains(" ") || s.Contains("$") || s.Contains("'"))
        {
            return $"'{s.Replace("'", "'\"'\"'")}'";
        }
        return s;
    }
}
